tools/shared.py

```diff
@@ -739,36 +738,38 @@ def has_vanilla_targets():
 
     def get_vanilla_file():
       logger.debug('regenerating vanilla file: %s' % LLVM_ROOT)
-      saved_file = os.path.join(temp_cache.dirname, 'is_vanilla.txt')
+      saved_file = Cache.get_path('is_vanilla.txt', root=True)
       if os.path.exists(saved_file):
         logger.debug('old: %s\n' % open(saved_file).read())
       open(saved_file, 'w').write(('1' if has_vanilla_targets() else '0') + ':' + LLVM_ROOT + '\n')
       return saved_file
 
-    is_vanilla_file = temp_cache.get('is_vanilla.txt', get_vanilla_file)
+    is_vanilla_file = Cache.get('is_vanilla.txt', get_vanilla_file, root=True)
     if CONFIG_FILE and os.path.getmtime(CONFIG_FILE) > os.path.getmtime(is_vanilla_file):
       logger.debug('config file changed since we checked vanilla; re-checking')
-      is_vanilla_file = temp_cache.get('is_vanilla.txt', get_vanilla_file, force=True)
+      is_vanilla_file = Cache.get('is_vanilla.txt', get_vanilla_file, force=True, root=True)
     try:
       contents = open(is_vanilla_file).read().strip()
       middle = contents.index(':')
       is_vanilla = int(contents[:middle])
       llvm_used = contents[middle + 1:]
       if llvm_used != LLVM_ROOT:
         logger.debug('regenerating vanilla check since other llvm (%s vs %s)`', llvm_used, LLVM_ROOT)
-        temp_cache.get('is_vanilla.txt', get_vanilla_file, force=True)
+        Cache.get('is_vanilla.txt', get_vanilla_file, force=True, root=True)
         is_vanilla = has_vanilla_targets()
     except Exception as e:
       logger.debug('failed to use vanilla file, will re-check: ' + str(e))
       is_vanilla = has_vanilla_targets()
-    temp_cache = None
     if is_vanilla:
       logger.debug('check tells us to use wasm backend')
       LLVM_TARGET = WASM_TARGET
     else:
       logger.debug('check tells us to use asm.js backend')
       LLVM_TARGET = ASM_JS_TARGET
 
+  if LLVM_TARGET == WASM_TARGET:
+    Settings.WASM_BACKEND = 1
```

Comment:
This was necessary to move here I assume? These code paths confuse me, but if the tests pass I'm confident.

Reply:
Yeah, this was happening during Settings creation before.. but now the settings are created earlier before we know the llvm version.